src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/VSMSBuildNuGetProjectSystem.cs

```diff
@@ -481,7 +481,22 @@ where AssemblyNamesMatch(assemblyName, assemblyReferenceNode.Item2)
                     string.Format(CultureInfo.CurrentCulture, Strings.FailedToAddReference, name), e);
             }
 
-            NuGetProjectContext.Log(ProjectManagement.MessageLevel.Debug, Strings.Debug_AddReference, name, projectName);
+            NuGetProjectContext.Log(
+                ProjectManagement.MessageLevel.Debug,
+                $"Added reference '{name}' to project:'{projectName}' resolvedToPackage:{resolvedToPackage} dteOriginalPath:{dteOriginalPath} assemblyFullPath:{assemblyFullPath}.");
+        }
+
+        private static bool IsSamePath(string path1, string path2)
+        {
+            if (string.IsNullOrWhiteSpace(path1)
+                || string.IsNullOrWhiteSpace(path2))
```

Comment:
Should this return `true` when both paths are empty or contain white-spaces only?